### PR TITLE
feat(api): add labware containment to enable loading the vacuum module collar on the dock

### PR DESCRIPTION
**Pull request by vegano1**
# Overview

Pr had to be re-created, I accidentally closed the original [Opentrons/opentrons#21299](https://github.com/Opentrons/opentrons/pull/21299)

This PR adds initial support in the load_labware protocol engine command for loading the Millipore vacuum manifold collar onto the vacuum module’s private staging dock. It introduces the foundational labware containment functionality needed to support nested workflows with the vacuum module.

### The vacuumModuleV1 provides two distinct addressable areas:

- vacuumModuleV1A3 — the main module base area
- vacuumModuleV1DockA4 — the private staging dock

This design allows labware to be loaded or moved to either location independently. With the new labware containment feature, an addressable area can hold multiple pieces of labware as long as their geometries do not collide. There are two kinds of relationships:

Parent relationship (-->) — normal loading/stacking (labware is placed on something)
Contained relationship (-.->) — geometric containment via `containedSpace` (labware sits inside another labware)

In the example below, you can see
- On the left, `Collar1` is loaded on the main module area (`vacuumModuleV1`). It contains the `plate` Labware inside its `containedSpace`, while also having `filter_plate1` Labware stacked on top of it at the same time.
- On the right, `Collar2` is loaded on the (`vacuumModuleV1DockA4`) addressable area, and then `filter_plate2` is loaded on top of the `Collar2` adapter.
- Both of these share the `vacuumModuleV1` Module and `CutoutFixture` parent location.

```mermaid
graph TD;

    Labware:filter_plate2-->|parent|Adapter:Collar2;
    Adapter:Collar2-->|parent|AddressableArea:vacuumModuleV1DockA4;
    Labware:filter_plate1-->|parent|Adapter:Collar1;
    Labware:plate-.->|contained|Adapter:Collar1;
    Labware:plate-->|parent|AddressableArea:vacuumModuleV1A3;
    Adapter:Collar1-->|parent|AddressableArea:vacuumModuleV1A3;
    AddressableArea:vacuumModuleV1A3-->|parent|Module:vacuumModuleV1;
    AddressableArea:vacuumModuleV1DockA4-->|parent|Module:vacuumModuleV1;
    Module:vacuumModuleV1-->|parent|CutoutFixture;
```

## Test Plan and Hands on Testing

The following protocol should simulate

```python
from opentrons.protocol_api import ProtocolContext

"""Load Vacuum Module."""

metadata = {
    "protocolName": "Vacuum Module API",
    "author": "Opentrons <protocols@opentrons.com>",
}
requirements = {
    "robotType": "Flex",
    "apiLevel": "2.28",
}

def run(ctx: ProtocolContext) -> None:
    """Protocol."""

    vm_mod = ctx.load_module( module_name="vacuumModuleV1", location="A3")
    black_flat_plate = vm_mod.load_labware("corning_96_wellplate_360ul_flat")
    manifold_collar = vm_mod.load_adapter('millipore_vacuum_manifold_collar_tall')
    white_filter_plate = manifold_collar.load_labware("invitroven_filter_plate")

```

## Changelog

- Added load_adapter_to_dock() method on VacuumModuleContext to conveniently load compatible adapters (such as the manifold collar) directly onto the dock.
- Introduced early labware containment support in the API layer to allow loading labware inside adapters that define `containedSpace`.
- Enabled loading the `millipore_vacuum_manifold_collar_tall` onto the vacuum module dock (location "A4" on Flex).
- Updated protocol execution flow to support collar + filter plate nesting scenarios.
- Minor updates to labware loading logic to handle the new dock addressable area.

### CODE REVIEW

**Review by sfoster1 — CHANGES_REQUESTED** [review]
there's a lot of new code in `geometry.py` and in `load_labware.py` and we really need tests for that new functionality. these are complex geometry calculations; we should probably have parametrized tests to check edge cases. it's one of those "probably more tests than code" cases.

File: api/src/opentrons/protocol_api/core/engine/deck_conflict.py [hunk 1]
```diff
@@ -246,14 +246,21 @@ def _map_labware(
...
     location_from_engine = engine_state.labware.get_location(labware_id=labware_id)
     if isinstance(location_from_engine, AddressableAreaLocation):
+        area_name = location_from_engine.addressableAreaName
+        # NOTE(ba, 2026-05-1): Some labware like the vacuum module collar
+        # are loaded onto custom addressable areas like `vacuumModuleV1DockV4`.
+        # Currently this does not map well to our convention of everything is
+        # either a DeckSlotName or StagingSlotName. So we need to take the
+        # last 2 characters to convert properly, we should find a better way
+        # of doing this like working with addressableAreaLocation directly.
+        area_name = area_name if len(area_name) == 2 else area_name[-2:]
```
Comment (sfoster1) [inline api/src/opentrons/protocol_api/core/engine/deck_conflict.py]: this makes me really uncomfortable. can we instead have this as another try/except or something? like if we can't build a `StagingSlotName` from it, try just the last two characters? could work with factoring out just parsing the area name into a deckslot or staging slot or something

**Review by sfoster1 — APPROVED** [review]
Thanks for the tests! Looks good to me, though I would like to find a better way of handling the string checks.

File: api/src/opentrons/protocol_api/core/engine/deck_conflict.py (hunk 1 above)
Comment (vegano1) [reply api/src/opentrons/protocol_api/core/engine/deck_conflict.py]: I added `_addressable_area_to_slot` helper to encapsulate this behavior.
